exercises/perfectShuffle.js

```diff
@@ -9,21 +9,17 @@
 
 function perfectShuffle(array) {
   // This is your job. :)
-  let firstHalfDeck = [];
-  let secondHalfDeck = [];
   let halfDeck = array.length/2;
   let shuffledDeck = [];
-  for (let i = 0; i < array.length; i++) {
-    if (i < halfDeck){
-      firstHalfDeck.push(array[i])
+  for (let i = 0; i < halfDeck; i++) {
+    shuffledDeck.push(array[i]);
+    if((halfDeck + i) < array.length){
```

Comment:
I'm not sure what role this check plays. The following should work:

```javascript
for (let i = 0; i < halfDeck; i++) {
  shuffledDeck.push(array[i]);
  shuffledDeck.push(array[i + halfDeck]);
}
```

When `i` is `halfDeck - 1` then `i + halfDeck` is `halfDeck + halfDeck - 1 === array.length - 1`.

```